tests/integration/framework/process/daprd/daprd.go

```diff
@@ -0,0 +1,177 @@
+/*
+Copyright 2023 The Dapr Authors
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package daprd
+
+import (
+	"context"
+	"errors"
+	"net"
+	"strconv"
+	"testing"
+
+	"github.com/google/uuid"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/dapr/dapr/tests/integration/framework/binary"
+	"github.com/dapr/dapr/tests/integration/framework/freeport"
+	"github.com/dapr/dapr/tests/integration/framework/process"
+	"github.com/dapr/dapr/tests/integration/framework/process/exec"
+)
+
+// options contains the options for running Daprd in integration tests.
+type options struct {
+	execOpts []exec.Option
+
+	appID                   string
+	appPort                 int
+	grpcPort                int
+	httpPort                int
+	internalGRPCPort        int
+	publicPort              int
+	metricsPort             int
+	profilePort             int
+	appHealthCheck          bool
+	appHealthCheckPath      string
+	appHealthProbeInterval  int
+	appHealthProbeThreshold int
+}
+
+// Option is a function that configures the dapr process.
+type Option func(*options)
+
+type Daprd struct {
+	exec     process.Interface
+	freeport *freeport.FreePort
+
+	appID            string
+	appPort          int
+	grpcPort         int
+	httpPort         int
+	internalGRPCPort int
+	publicPort       int
+	metricsPort      int
+	profilePort      int
+}
+
+func New(t *testing.T, fopts ...Option) *Daprd {
+	t.Helper()
+
+	uid, err := uuid.NewUUID()
```

Comment:
Is there a reason for using `NewUUID` (UUID v1) rather than `New` (UUID v4)?

Reply:
Fixed